docs/designs/credential.md

```diff
@@ -0,0 +1,64 @@
+# Credential
+
+This document specifies how DocFX accesses private resoruces during restore/build/publish. 
+
+The credential may be used to:
+- clone a private repository
+- download resource from a private blob (e.g. [GitHub user cache](github-user-cache.md))
+- update a resource, also like GitHub user cache
+- publish files
+
+## How to authorize
+
+In general, DocFX uses standard HTTP request methods to access resources, so that it does not care about the hosting details about the resources. 
+- `GET`: to fetch resource
+- `PUT`: to upload resource
+
+DocFX need to embed the credential information in the HTTP request. URL query and request header can be the right place.
+
+### URL query
+
+URL can embed the credential in itself query string. e.g.
+```
+https://contoso.com/path/file?sig={token}
+```
+
+It can be configured in `docfx.yml`:
+```yml
+referenceToAResource: "https://contoso.com/path/file"
+secrets:
+  "https://contoso.com/path/file": "?sig={token}"
+```
+The value of secrets object can be a simple string, which indicates the param string. 
+
+In this case, we can also embed the token into URL like `referenceToAResource: "https://contoso.com/path/file?sig={token}"`. However, the config cannot be public as it contains secret infomation. Have a separate `secrets` part can help store them in more private place, like a local machine.
+
+### Request header
+
+Request header can also contains credential in `Authorization` header. e.g.
+```
+Authorization: Bearer {token}
+```
+The value of secrets object can be an object to provide more details of the request header. It can be configured in `docfx.yml` like:
+```yml
+secrets:
```

Comment:
there will be other secrets that are not determined by base URL, like GitHub access token, app insight credential, etc.